umbraco/Umbraco-CMS
# Netcore: Package support for media + media types

### Details
- Implements support for media and media types when creating packages
- Implements support for loading media+media types from packages
- Fixed issue with logout just before restart

### Notes
To not duplicate massive amount of code, I had to make lots of the content/document methods generic, to work for both `IContent` + `IMedia` and `IContentType` + `IMediaType`

### Test
- Go to backoffice 
- Create custom media types as copies of image and folder
- Create media and folders with media of the new types.
- Go to the package section and choose the "Created" content app
- Create a package with media and media types
- Ensure you can edit the package in the UI
- Download the package
   - Test with and without child media included
- Remove the media selected for the file
- Remove the custom media types
- Try install the downloaded package, and ensure the media and media types are created
